### PR TITLE
feat: add preload support

### PR DESCRIPTION
**Pull request by maksg**
## Description

Adds support for using `CommonActions.preload`.
Also can be used to for iOS PiP https://github.com/software-mansion/react-native-screens/pull/1665.

Corresponding PR in `react-navigation`:

* https://github.com/react-navigation/react-navigation/pull/12083

## Changes

- Added prop `hiddenFromStack`

## Test code and steps to reproduce

1. Run new test `TestPreload`.
2. Tap on preload buttons and check if the preload time increases.

## Checklist

- [x] Included code example that can be used to test this change
- [x] Updated TS types
- [x] Updated documentation: 
  - [x] https://github.com/software-mansion/react-native-screens/blob/main/guides/GUIDE_FOR_LIBRARY_AUTHORS.md
  - [x] https://github.com/software-mansion/react-native-screens/blob/main/native-stack/README.md
  - [x] https://github.com/software-mansion/react-native-screens/blob/main/src/types.tsx
  - [x] https://github.com/software-mansion/react-native-screens/blob/main/src/native-stack/types.tsx
- [ ] Ensured that CI passes

### CODE REVIEW

File: src/types.tsx [hunk 1]
```diff
@@ -161,6 +161,17 @@ export interface ScreenProps extends ViewProps {
...
   homeIndicatorHidden?: boolean;
+  /**
+   * If a screen is marked `hidden`, the resources will be retained, but it will not be rendered
+   * in the stack. This is used when preloading screen but can also be used as a way to support
+   * iOS picture-in-picture, where you have a video overlay originating on a screen, and even
+   * when popping the screen off the stack the resources need to be retained (not garbage collected),
+   * so that the video can continue playing. Additionally, this allows the screen to be "un-hidden"
+   * (visually, pushed back onto the stack) when the user taps on the overlay video.
+   *
+   * Only supported on iOS and Android.
```
Comment (WoLewicki) [inline src/types.tsx:172]: If it is supported on `Android`, should the code for `Android` end up in this PR?

Comment (maksg) [reply src/types.tsx:172]: I would like to add that to this PR once I'm sure the iOS side works as expected. But it could be split up if you want.

Comment (WoLewicki) [reply src/types.tsx:172]: I think the PR is pretty small so its ok to have both platforms in one PR.

**Review by kkafar — COMMENTED** [review]
Lemme know once the Android code is also ready

**Comment by maksg** [comment]
@kkafar Android is now ready

**Comment by fobos531** [comment]
> @kkafar Android is now ready

@maksg Might be a good idea to rename the PR to indicate that Android support is also present in this PR.

**Pull request closed without merging**